### ISSUE TITLE
current github branch signature-validate (05-May-2021 10:00 CET) --> could only sign once

### ISSUE THREAD

**Issue by aheider** [LibreSign/libresign#222]
You made great progress so far! Thank you for this!

To the issue:
- I was able to download the source and build with yarn, although yarn got stuck at "+ 614 hidden modules".
- moved files into app folder
- de- and reactivated the app
- checked LibreSign NC app --> empty page
- checked files details tab of a PDF in NC20 --> tab "sign" = "unterschreiben" in German
- selected a *.pfx for signing, used the password --> worked! I got a checkmark!
- a new PDF "*_signed.pdf" was created in the correct folder --> downloaded the PDF
- in the PDF reader (Adobe Reader) the signature was there
- as I have added the CFSSL root CA to trusted hosts for signing the file and the signature were found valid and I could see the correct user (email address) which signed the file

Great Job!

I then went on and tried another PDF file --> "sign" tab was gone in the file details tab

Does there something maybe need to be reset to work?

Thank You again!

**Comment by vitormattos** [comment]
We are migrating from npm to yarn in #206
The implementation of this tab does not work in Nextcloud 21. In #206, a new LibreSign tab is being implemented.

**Comment by aheider** [comment]
Maybe there was misunderstanding:
- I used yarn, not npm for building
- I used current git not a released code package
- I used branch "signature-validation" not main branch
- I run NC20 and not NC21
- I was successfully able to use the files tab for signing (which didn't work with release 2.2.2)

The issue was just:
- signing was only possible ONCE, second time around the files tab for signing was gone
- I did not change anything in between
- I think this problem is a bug in the new files tab code

Hope that helps.

Thank you.

**Comment by vitormattos** [comment]
Very grateful for the tests and feedbacks!

We are currently working on exactly this point in the code.

In the N-1 release, one of the objectives is to have this tab 100% working. The N-1 is almost finished.

At N-2 we will implement a CRUD on the LibreSign icon to manage requests to sign and various code improvements.

https://github.com/LibreSign/libresign/projects

PS: we created an account on [Patreon](https://patreon.com/librecode). It is already possible to use but we are still going to improve.

We also created a [Bountysource](https://www.bountysource.com/teams/libresign/issues) account

**Comment by vinicios-gomes** [comment]
Hi @aheider,
We recently completed the correction of some bugs in the signature-validity branch.
It is stable to the tests, we await your feedback.
On behalf of the entire @LibreSign/developer team, we appreciate your feedback.

PS: we created an account on [Patreon](https://patreon.com/librecode). It is already possible to use.

We also created a [Bountysource](https://www.bountysource.com/teams/libresign/issues) account
